packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts

```diff
@@ -414,6 +462,70 @@ export function useUpdateEntityACL(
   })
 }
 
+export function useDeleteEntityACL(
+  options?: Partial<UseMutationOptions<void, SynapseClientError, string>>,
+) {
+  const queryClient = useQueryClient()
+  const { accessToken, keyFactory } = useSynapseContext()
+
+  return useMutation<void, SynapseClientError, string>({
+    ...options,
+    mutationFn: (entityId: string) =>
+      SynapseClient.deleteEntityACL(entityId, accessToken),
+    onSuccess: async (result: void, entityId, ctx) => {
+      await onMutateEntityAclSuccess(entityId, null, queryClient, keyFactory)
+
+      if (options?.onSuccess) {
+        await options.onSuccess(result, entityId, ctx)
+      }
+    },
+  })
+}
+
+function useGetEntityBenefactorACLQueryOptions(
+  entityId: string,
+): UseQueryOptions<
+  EntityBundle<{ includeBenefactorACL: true }>,
+  SynapseClientError,
+  AccessControlList
+> {
+  const opts = useGetEntityBundleQueryOptions<{ includeBenefactorACL: true }>(
+    entityId,
+    undefined,
+    {
+      includeBenefactorACL: true,
+    },
+  )
+
+  return {
+    ...opts,
+    select: data => data.benefactorAcl,
+  }
+}
+
+/**
+ * Retrieve the ACL of an entity. This call will succeed even for entities where the caller
+ * does not have READ permission.
+ * @param entityId
+ * @param options
+ */
+export function useSuspenseGetEntityBenefactorACL(
```

Comment:
Created a dedicated hook for fetching the `benefactorACL` via the entity bundle. This is valuable because this call will always succeed regardless of your permissions on an entity. 

If you were to request any additional bundle fields on an entity where you do not have READ access, the call will fail.

This is useful for the case where you can CHANGE_PERMISSIONS on an entity but you cannot READ the parent container. You can still remove local sharing settings on the file and see the parent benefactor ACL using this call.

Reply:
Interesting!